openhtf/plugs/usb/__init__.py

```diff
@@ -36,56 +38,153 @@ def MyPhase(test, adb):
 conf.Declare('usb_server', 'USB Server IP/Hostname')
 conf.Declare('usb_server_port', 'USB Server Port',
              default_value=10000)
-
 conf.Declare('libusb_rsa_key', 'A private key file for use by libusb auth.')
+conf.Declare('device_type', 'android, fastboot or general usb type')
+conf.Declare('usb_hub', 'local or Ethersyncy')
+
+_LOG = logging.getLogger(__name__)
+
+class GeneralUsbAttributeError(Exception):
+  """Indicates the usb class/subclass/protocol info was invalid."""
+
+class general_usb(object):
+  def __init__(self, usb_class=None, usb_subclass=None, usb_protocol=None):
+    self.usb_class = usb_class
+    self.usb_subclass = usb_subclass
+    self.usb_protocol = usb_protocol
+
+  @property
+  def CLASS(self):
+    return self.usb_class
+  @property
+  def SUBCLASS(self):
+    return self.usb_subclass
+  @property
+  def PROTOCOL(self):
+    return self.usb_protocol
+
+""" Config file example for device at Ethersync
+device_type: android (fastboot, general)
+usb_hub: 
+  Ethersync:
+   unit_name: EtherSyncca9167
+   mac_addr: 78:a5:04:ca:91:66
+   port_addr: 8589934803
+
+---- Config file example for android/fastboot usb at local 
+device_type: android (fastboot)
+usb_hub: local
+
+---- Config file example for android/fastboot usb at local
+device_type: general
+usb_hub: 
+   local:
+      class: xxxxx
+      subclass: xxxxx
+      protocol: xxxxx
+      vendor_id:  xxxx (optional)
+      product_id: xxxxx(optional)
+      serial: (device serial number, optional)
+"""
 
 
-def _open_usb_handle(**kwargs):
-  """Open a UsbHandle subclass, based on configuration.
-
-  If configuration 'usb_server' is set, use it to connect to remote usb,
-  otherwise attempt to connect locally.
-
-  Args:
-    **kwargs: Arguments to pass to respective handle's Open() method.
-
-  Returns:
-    Instance of UsbHandle.
-  """
-  if conf.Config().usb_server:
-    return None
+def _get_usb_serial(port_addr):
+  """Get a usb serial based on the Cambrionix unit mac address in configuration."""
+  cmd = '/usr/local/google/home/amyxchen/esuit64 -t "DEVICE INFO, %s"' % port_addr
```

Comment:
This brings up a good point. esuit64 is the tool we have for managing EtherSync connections. Do we make having it in the executable path a requirement for this plug? What happens if the user has esuit32 instead (if such a thing exists)? It might be worth asking Cambrionix again about their python package for EtherSync since it has been a couple of months since we last heard from them about it.